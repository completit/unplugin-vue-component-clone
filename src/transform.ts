export function transformCode(code: string) {
  return code;
}
