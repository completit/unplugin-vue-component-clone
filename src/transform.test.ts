import { describe, it, expect } from "vitest";
import { transformCode } from "./transform";

describe("transformCode", () => {
  it("should transform code", () => {
    const code = `
    const render = (_ctx, _cache) => {
      const _component_el_infinite_scroll = _resolveComponent("el-infinite-scroll")
      const _directive_el_infinite_scroll = _resolveDirective("el-infinite-scroll")

      return _withDirectives(
        (_openBlock(),
        _createBlock(_component_test_comp, null, null, 512 /* NEED_PATCH */)),
        [[_directive_loading, 123]]
      )
    }`;
    const result = transformCode(code);
    expect(result).toMatchSnapshot();
  });
});
