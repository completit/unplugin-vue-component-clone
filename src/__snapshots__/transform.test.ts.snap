// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`transformCode > should transform code 1`] = `
"/* unplugin-vue-components disabled */import __unplugin_directives_0 from 'test/directive/Loading';
import __unplugin_components_0 from 'test/component/TestComp';
    const render = (_ctx, _cache) => {
      const _component_el_infinite_scroll = _resolveComponent("el-infinite-scroll")
      const _directive_el_infinite_scroll = _resolveDirective("el-infinite-scroll")

      return _withDirectives(
        (_openBlock(),
        _createBlock(_component_test_comp, null, null, 512 /* NEED_PATCH */)),
        [[_directive_loading, 123]]
      )
    }"
`;
